Source/WebCore/platform/graphics/gstreamer/ImageDecoderGStreamer.cpp

```diff
@@ -318,131 +297,63 @@ void ImageDecoderGStreamer::notifySample(GRefPtr<GstSample>&& sample)
     }
 }
 
-void ImageDecoderGStreamer::InnerDecoder::handleMessage(GstMessage* message)
+void ImageDecoderGStreamer::InnerDecoder::run()
 {
-    ASSERT(&m_runLoop == &RunLoop::current());
-
-    auto scopeExit = makeScopeExit([protectedThis = WeakPtr { *this }] {
-        if (!protectedThis)
-            return;
-        Locker locker { protectedThis->m_messageLock };
-        protectedThis->m_messageDispatched = true;
-        protectedThis->m_messageCondition.notifyOne();
+    Vector<uint8_t> data { m_buffer };
+    auto bufferSize = data.size();
+    auto bufferData = data.data();
+    auto buffer = gst_buffer_new_wrapped_full(GST_MEMORY_FLAG_READONLY, bufferData, bufferSize, 0, bufferSize, new Vector<uint8_t>(WTFMove(data)), [](gpointer data) {
```

Comment:
If you keep the constructor + `run` scheme, as I mention above, you should turn the span into a vector in the constructor already, otherwise the data ownership is not clear. And also the Thread + lambda thing.

Reply:
The data is safe because we wait the decoding completion anyway.